tests/unit/generateExpiryDate.unitTest.ts

```diff
@@ -1,461 +1,1846 @@
+              });
+          });
+        });
+        describe("with no previous expiry date, and today is 2 months before registration anniversary", () => {
+          beforeAll(() => {
+            dateMockUtils.setupDateMock("2020-02-01T10:00:00.000Z");
+          });
+          afterAll(() => {
+            dateMockUtils.restoreDateMock();
+          });
 
-                    const expectedExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1);
-                    return testResultsService.generateExpiryDate(hgvTestResult)
-                        .then((hgvTestResultWithExpiryDate: any) => {
-                            expect((hgvTestResultWithExpiryDate.testTypes[0].testExpiryDate).split("T")[0]).toEqual(expectedExpiryDate.toISOString().split("T")[0]);
-                        });
-                });
+          it("should set the expiryDate to 1 day prior to one year from \"now\" for Pass results - not for PRS and Fail", () => {
+            const test: ITestResult = cloneDeep(testResultsMockDB[0]);
+            test.regnDate = new Date("2019-04-01");
+            test.testTypes.forEach((type) => {
+              delete type.testExpiryDate;
+              delete type.testAnniversaryDate;
+              return type;
+            });
+            const psvTestResult = test;
+            const getBySystemNumberResponse = test;
+
+            MockTestResultsDAO = jest.fn().mockImplementation(() => {
+              return {
+                getBySystemNumber: () => {
+                  return Promise.resolve({
+                    Items: Array.of(getBySystemNumberResponse),
+                    Count: 1,
+                    ScannedCount: 1
+                  });
+                },
+                getTestCodesAndClassificationFromTestTypes: () => {
+                  return Promise.resolve({
+                    linkedTestCode: "wde",
+                    defaultTestCode: "bde",
+                    testTypeClassification: "Annual With Certificate"
+                  });
+                }
+              };
             });
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            const expectedExpiryDate = new Date("2021-01-31");
+            return testResultsService.generateExpiryDate(psvTestResult)
+              .then((psvTestResultWithExpiryDateAndTestNumber: any) => {
+                expect((psvTestResultWithExpiryDateAndTestNumber.testTypes[0].testExpiryDate).split("T")[0]).toEqual(expectedExpiryDate.toISOString().split("T")[0]);
+                expect(psvTestResultWithExpiryDateAndTestNumber.testTypes[1].testExpiryDate).toBeUndefined();
+                expect(psvTestResultWithExpiryDateAndTestNumber.testTypes[2].testExpiryDate).toBeUndefined();
+              });
+          });
         });
+        // describe("with no previous expiry date, and today is <2 months before registration anniversary", () => {
```

Comment:
You still need to fix these tests?

Reply:
The tests are fine - the functionality is in another ticket